Add forward flag to subscribe/subscribe_update (#743)
moq-wg/moq-transport: draft-ietf-moq-transport.md

both messages at the same time or wait for the first to return before sending
the second.

Subscribers can also request a publisher to not forward objects for a given

> track by setting the `Forward` field to 0. Relays SHOULD always set the `Forward`

Reviewer: ```suggestion
track by setting the `Forward` field to 0. This can reduce the latency of beginning delivery of
a track that is expected to be needed soon when a Relay is substantially closer to the
subscriber than the Original Publisher. Relays SHOULD always set the `Forward`
```

> flag to 1 when forwarding the SUBSCRIBE message upstream, regardless of the
> value of the `Forward` field from the downstream subscription.

Reviewer: ```suggestion
flag to 1 if a new subscription needs to be sent upstream, regardless of the
value of the `Forward` field from the downstream subscription. Subscriptions that
are not forwarded consume resources from the publisher, so a publisher might
deprioritize, reject, or close those subscriptions to ensure other subscriptions can
be delivered.
```

both messages at the same time or wait for the first to return before sending
the second.

> Subscribers can also request a publisher to not forward objects for a given

Reviewer: Is this clearer?

```suggestion
A Subscription can also request a publisher to not forward Objects for a given
```

> Subscribers can also request a publisher to not forward objects for a given

Reviewer 2: Yes, this is an improvement. And I guess talking about the subscription ensures that both Subscribe and Subscribe_update works.

track by setting the `Forward` field to 0. This can reduce the latency of receiving

> Objects that might be needed urgently when the Relay is substantially closer to the

Reviewer: No, it's Objects in the track being subscribed to, but those Objects are not currently delivered.

> Objects that might be needed urgently when the Relay is substantially closer to the

Reviewer: Would this change help?

```suggestion
Objects that might later be needed urgently when the Relay is substantially closer to the
```

> Objects that might be needed urgently when the Relay is substantially closer to the

Reviewer 2: I think part of the problem is this part of the sentence: "when the Relay is substantially closer". Closer in which sense I think I interpreted it from topological rather than transmission order. I would rewrite the the whole sentence:

This can reduce the latency until receiving
Objects when the Subscriber later urgently needs Objects for this track as the relay will have them available, rather than having to request them from an upstream relay or the Original publisher.

> Objects that might be needed urgently when the Relay is substantially closer to the

Reviewer: I started with your sentence and shortened it some, PTAL

both messages at the same time or wait for the first to return before sending
the second.

A Subscription can also request a publisher to not forward Objects for a given
track by setting the `Forward` field to 0. This can reduce the latency until
receiving Objects when the Subscriber later urgently needs them, because the
Relay will have them, rather than having to subscribe upstream. Relays SHOULD

> always set the `Forward` flag to 1 if a new subscription needs to be sent

Reviewer: ```suggestion
set the `Forward` flag to 1 if a new subscription needs to be sent
```

> A Subscription can also request a publisher to not forward Objects for a given
> track by setting the `Forward` field to 0. This can reduce the latency until
> receiving Objects when the Subscriber later urgently needs them, because the
> Relay will have them, rather than having to subscribe upstream. Relays SHOULD
> set the `Forward` flag to 1 if a new subscription needs to be sent
> upstream, regardless of the value of the `Forward` field from the downstream
> subscription. Subscriptions that are not forwarded consume resources from the
> publisher, so a publisher might deprioritize, reject, or close those
> subscriptions to ensure other subscriptions can be delivered.

Reviewer 3: ```suggestion
A Subscription can also request a publisher to not forward Objects for a given
track by setting the `Forward` field to 0.  This allows the publisher or relay
to  prepare to serve the subscription in advance, reducing the time to
receive objects in the future. Relays SHOULD set the `Forward` flag to 1 if a
new subscription needs to be sent upstream, regardless of the value of the 
`Forward` field from the downstream subscription. Subscriptions that are not
forwarded consume resources from the publisher, so a publisher might
deprioritize, reject, or close those subscriptions to ensure other
subscriptions can be delivered.
```

Reviewer: LG

  StartObject (i),
  EndGroup (i),
  Subscriber Priority (8),

>   Forward(8),

Reviewer: ```suggestion
  Forward (8),
```
